### PR TITLE
128T release notes 5.0.0

### PR DESCRIPTION
**Pull request by MichaelBaj**

### CODE REVIEW

File: docs/cli_reference.md [hunk 1]
```diff
@@ -4319,9 +4608,65 @@ Wed 2020-04-22 14:43:43 UTC
...
+| ---- | ----------- |
+| force | Skip confirmation prompt. Only required when targeting all routers |
+| node | node for which to display domain-name categories of active sessions |
+| router | router for which to display domain-name categories of active sessions |
+
+##### See Also
+
+| command | description |
+| ------- | ----------- |
+| [`show domain-names`](#show-domain-names) | Display app-id-v2 domain-names used by sessions |
```
Comment (ptimmons) [inline docs/cli_reference.md:4633]: no punctuation

File: docs/cli_reference.md [hunk 2]
```diff
@@ -4319,9 +4608,65 @@ Wed 2020-04-22 14:43:43 UTC
...
+
+| name | description |
+| ---- | ----------- |
+| request-order | Get domains sorted by most-sessions or most-recent |
+
+##### See Also
+
+| command | description |
+| ------- | ----------- |
+| [`show domain-categories`](#show-domain-categories) | Display app-id-v2 domain-name categories used by sessions |
```
Comment (ptimmons) [inline docs/cli_reference.md:4665]: No punctuation

File: docs/cli_reference.md [hunk 3]
```diff
@@ -4753,9 +5099,173 @@ intf10.0 gateway 172.16.12.1
...
+
+```
+show lte [device-interface <device-interface>] [force] [node <node>] router <router>
+```
+
+##### Keyword Arguments
+
+| name | description |
+| ---- | ----------- |
+| device-interface | lte device interface (default: all) |
```
Comment (ptimmons) [inline docs/cli_reference.md:5116]: Should probably capitalize LTE throughout this section.

File: docs/cli_reference.md [hunk 4]
```diff
@@ -6379,15 +6964,20 @@ Outputs a concise table with statistics on the RIB
...
+show rib summary [rows <rows>] [force] [router <router>]
 ```
 
 ##### Keyword Arguments
 
 | name | description |
 | ---- | ----------- |
 | force | Skip confirmation prompt. Only required when targeting all routers |
-| router | The name of the router for which to display the RIB summary |
+| router | The name of the router for which to display the RIB summary (default: &lt;current router&gt;) |
```
Comment (ptimmons) [inline docs/cli_reference.md:6975]: inconsistent Capitalization in these rows.

File: docs/cli_reference.md [hunk 5]
```diff
@@ -6458,6 +7048,61 @@ Completed in 0.17 seconds
...
+
+```
+show session-captures [{id <id> | detail}] [service <service>] [force] [router <router>] [node <node>]
+```
+
+##### Keyword Arguments
+
+| name | description |
+| ---- | ----------- |
+| detail | display session-captures in detail |
```
Comment (ptimmons) [inline docs/cli_reference.md:7091]: Inconsistent capitalization

File: docs/cli_reference.md [hunk 6]
```diff
@@ -6580,6 +7225,56 @@ Completed in 0.10 seconds
...
+
+```
+show step lsdb [originator <originator-name>] [force] router <router> [<verbosity>]
+```
+
+##### Keyword Arguments
+
+| name | description |
+| ---- | ----------- |
+| force | Skip confirmation prompt. Only required when targeting all routers |
```
Comment (ptimmons) [inline docs/cli_reference.md:7242]: Inconsistent capitalization

File: docs/cli_reference.md [hunk 7]
```diff
@@ -6580,6 +7225,56 @@ Completed in 0.10 seconds
...
+| ip-prefix | STEP routes for this ip prefix [type: IP prefix] |
+| node | STEP routes on this node |
+| router | the router to request STEP information from |
```
Comment (ptimmons) [inline docs/cli_reference.md:7269]: Initial caps

**Review by ptimmons — CHANGES_REQUESTED** [review]

File: docs/release_notes_128t_5.0.md [hunk 8]
```diff
@@ -1,135 +1,98 @@
...
+- **I95-9152** PCLI configuration help text now indicates which fields are required
 ------
-- **I95-12833** PCLI should discourage users from modifying deprecated config fields
+- **I95-9242** Service and Topology Exchange Protocol (STEP)
 ------
-- **I95-15618** Add completers to the "clear arp" PCLI command
+- **I95-15618** Auto completion now available on `clear arp` PCLI command
 ------
-- **I95-21776** Ethernet over SVR
+- **I95-20757** Displaying alarm history within the PLCI via `show events alarm` now supports relative timestamps
```
Comment (ptimmons) [inline docs/release_notes_128t_5.0.md:16]: PCLI

File: docs/release_notes_128t_5.0.md [hunk 9]
```diff
@@ -1,135 +1,98 @@
...
-- **I95-36646** SVR Savings: Page continues to refresh on its own, resets router selector on refresh
+- **I95-36828** Unable to acquire logs through GUI when `remote-login` is disabled
 
 ## Special Considerations
 
-- **I95-34624** Remove AP driven install from 128T
+- **I95-12833** Provisioning deprecated configuration fields now prompts for confirmation
 ------
-- **I95-34983, I95-35892** Remove unused PCLI Commands
+- **I95-20718** CLI now produces a warning when creating configuration objects with the keywords ("delete", "force", "move", "clone", "all") as their name
```
Comment (ptimmons) [inline docs/release_notes_128t_5.0.md:87]: PCLI

File: docs/release_notes_128t_5.0.md [hunk 10]
```diff
@@ -1,135 +1,98 @@
...
 ------
-- **I95-35609** Enhanced software delivery mechanism (Artifactory and changes to file naming)
+- **I95-34624** Remove AP driven install from 128T
```
Comment (ptimmons) [inline docs/release_notes_128t_5.0.md:89]: I recommend spelling out Automated Provisioner

**Review by ptimmons — CHANGES_REQUESTED** [review]

File: docs/about_releases.md [hunk 11]
```diff
@@ -5,7 +5,8 @@ sidebar_label: 128T Releases
 
 | Version                                          | Release            | GA Date            | End of Maintenance | End of Support     |
 | ------------------------------------------------ | ------------------ | ------------------ | ------------------ | ------------------ |
-| [5.0.0](release_notes_128t_5.0.md#release-5.0)   | October 16, 2020   |                    | April 16, 2021     | October 16, 2021   |
```
Comment (MichaelBaj) [inline docs/about_releases.md]: Looks like 5.0.0 was removed.

File: docs/release_notes_128t_5.0.md [hunk 12]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
-- **I95-22789** Dual LTE Support
+- **I95-27886 Packet Duplication for inter-node High Availability:** Packet duplication over multiple inter-node links helps reduce packet loss during transmission. For protocols such as UDP that do not verify packet integrity, this helps ensure full transmission of traffic.
 ------
-- **I95-24412** Dual PPPoE Support
+- **I95-28531 Application Categorization:** Application Identification/Categorization utilizes a database of traffic categories, such as “News”, “Social Media”, “Health”, “Sports”, etc. These categories can be assigned to services, which can then be prioritized, monitored for traffic levels, or blocked for security purposes. Each category is populated with a list of domain names associated with the type of traffic.
 ------
-- **I95-27886** Packet Duplication for inter-node HA
+- **I95-31218 Configurable BFD Hold-Down Timer:**
+
+
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md]: extra line breaks

File: docs/release_notes_128t_5.0.md [hunk 13]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
+------
+- **I95-31754 PCLI Log Level:**
+
+
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md]: Empty description and extra line breaks

File: docs/release_notes_128t_5.0.md [hunk 14]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
 ------
-- **I95-28531** Application Categorization
+- **I95-32558 In-line Flow Performance Monitoring:** Inline performance metrics have been enhanced to achieve a finer granularity than what is currently available via BFD. Statistics are now collected per path, priority, and protocol (TCP or UDP) level. Sessions can be forwarded using more granular SLA metrics.
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md:38]: I believe it is "per path, priority, and traffic class".  Please double check with Scott McCulley.

File: docs/release_notes_128t_5.0.md [hunk 15]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
-- **I95-34823** About page includes link to online documentation
+- **I95-35150 [PCLI Clone Configuration](cli_reference.md#clone):** The clone command duplicates the configuration data from an existing router into a new router with a new name, and stages it to the candidate configuration.
 ------
-- **I95-35150** [PCLI Clone Configuration](cli_reference.md#clone)
+- **I95-35190 [PCLI bulk paste configuration](concepts_pcli.md#paste-config):** The PCLI detects configuration entered in bulk and accepts input in either show config native format or flat format.
 ------
-- **I95-35190** [PCLI bulk paste configuration](concepts_pcli.md#paste-config)
+- **I95-35212 Adaptive Encryption Improvements:** 
+
+
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md]: Extra empty lines

File: docs/release_notes_128t_5.0.md [hunk 16]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
 ------
-- **I95-35741** [PCLI plugin service commands](cli_reference.md#manage-plugin-install)
+- **I95-35741 [PCLI plugin service commands](cli_reference.md#manage-plugin-install):** Plugin installation and removal is handled through the PCLI. 
+
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md]: missing line break between entries

File: docs/release_notes_128t_5.0.md [hunk 17]
```diff
@@ -5,48 +5,71 @@ sidebar_label: '5.0'
...
-- **I95-35150** [PCLI Clone Configuration](cli_reference.md#clone)
+- **I95-35190 [PCLI bulk paste configuration](concepts_pcli.md#paste-config):** The PCLI detects configuration entered in bulk and accepts input in either show config native format or flat format.
 ------
-- **I95-35190** [PCLI bulk paste configuration](concepts_pcli.md#paste-config)
+- **I95-35212 Adaptive Encryption Improvements:** 
+
+
 ------
-- **I95-35741** [PCLI plugin service commands](cli_reference.md#manage-plugin-install)
+- **I95-35741 [PCLI plugin service commands](cli_reference.md#manage-plugin-install):** Plugin installation and removal is handled through the PCLI.
```
Comment (MichaelBaj) [inline docs/release_notes_128t_5.0.md:56]: Suggested rewording:
Plugin lifecycle management can now be managed through the PCLI.

File: sidebars.js [hunk 18]
```diff
@@ -66,11 +66,13 @@ module.exports = {
           "config_dns_proxy",
           "config_ldap",
           "config_management_over_forwarding",
+
```
Comment (MichaelBaj) [inline sidebars.js:69]: Why was this empty line added?

File: sidebars.js [hunk 19]
```diff
@@ -66,11 +66,13 @@ module.exports = {
...
           "config_ha",
           "config_dual_router_ha",
           "config_non_forwarding_ha_interfaces",
           "config_adding_interfaces_to_ha_team",
           "config_transition_standalone_to_ha",
+
```
Comment (MichaelBaj) [inline sidebars.js:75]: Why was this empty line added?

File: docs/release_notes_128t_5.0.md (hunk 17 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_5.0.md:56]: edit made.

File: docs/release_notes_128t_5.0.md (hunk 16 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_5.0.md]: resolved

File: docs/release_notes_128t_5.0.md (hunk 15 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_5.0.md]: resolved

File: docs/about_releases.md (hunk 11 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/about_releases.md]: resolved

File: sidebars.js (hunk 19 above)
Comment (Chr1st0ph3rTurn3r) [reply sidebars.js:75]: I add spaces to keep track of what I have re-organized. The spaces don't show up in the published docs, so if anything it is for readability in the markdown.

File: docs/release_notes_128t_5.0.md (hunk 14 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_5.0.md:38]: We were both wrong. I got the correct info from Scott and am adding it.

File: docs/about_releases.md [hunk 20]
```diff
@@ -5,6 +5,7 @@ sidebar_label: 128T Releases
 
 | Version                                          | Release            | GA Date            | End of Maintenance | End of Support     |
 | ------------------------------------------------ | ------------------ | ------------------ | ------------------ | ------------------ |
+| [5.0.0](release_notes_128t_5.0.md#release-5.0)   | October 16, 2020   |                    | April 16, 2021     | October 16, 2021   |
```
Comment (MichaelBaj) [inline docs/about_releases.md:8]: Release date needs to be updated to 12/18 (at least that's our current projected date)

File: docs/sec_adaptive_encrypt.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
+---
+title: Adaptive Encryption
+sidebar_label: Adaptive Encryption
+---
+
+## Introduction
+
+In order to provide security across the WAN, the 128T Networking Platform provides the functionality to encrypt between peered routers. However, for traffic that is already encrypted by the end host, this is redundant. Adaptive Encryption is aimed at identifying the encrypted traffic and providing security for non-encrypted traffic by encrypting it through the 128T Router.
+
+On a service where most of the traffic is expected to be encrypted, the goal is to identify and encrypt those few sessions that are not. This eliminates the performance penalty of encrypting traffic that is already encrypted.
```
Comment (MichaelBaj) [inline docs/sec_adaptive_encrypt.md:10]: suggest changing "penalty" to "overhead"

File: docs/sec_adaptive_encrypt.md [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+## Non-TCP and Non-IPSEC Protocols
+
+If a packet is plain UDP, ICMP, or any other non-TCP and non-IPSEC protocol, it is assumed to be unencrypted and is installed with encryption actions. 
+
+## Configuration
+
+Adaptive Encryption is enabled by a boolean in the [security-policy](config_reference_guide.md#security):
+
+```
+leaf adaptive-encryption {
```
Comment (MichaelBaj) [inline docs/sec_adaptive_encrypt.md:42]: Please swap out the yang with actual configuration example.

File: docs/howto_tune_bfd.md [hunk 23]
```diff
@@ -71,6 +71,57 @@ In other words, "*if I don't receive a message in the amount of time that is my
...
+
+#### Parameters
+
+**Initial Hold Down Timer:** The minimum amount of time BFD must wait before beginning notifications. The default is 5 seconds, and can be configured to be any value lower than the `maximum-hold-down-time`. The hold-down-timer will not accept a value of 0. 
+
+**Maximum Hold Down Timer:** The maximum amount of time that BFD must wait before it begins notifications. This timer only applies when BFD damping is enabled. The default value for this is 3600 s (or 1 hour). The network administrator may configure this to be any value higher than the initial`hold-down-time`.
+
+**Dynamic Damping:** Disabled by default. When enabled and the initial `hold-down-time` and `maximum-hold-down-time` values are configured, the 128T routers dynamically adjust the values to ensure that BFD flaps are not affecting the system. This prevents the effect of oscillations, or flapping, caused by BFD and underperforming (or volatile) links. It ensures stability of the entire network and reduces the events requiring the network administrator's attention.
```
Comment (robert-patterson) [inline docs/howto_tune_bfd.md:88]: `hold-down-time` and `maximum-hold-down-time` have default values, so it is not strictly necessary for the user to configure them for the dynamic adjustment to be applied.

When enabled, the 128T router uses the `hold-down-time` and `maximum-hold-down-time` parameters to dynamically adjust the damping timer to ensure that excessive BFD flaps are not affecting the system negatively.

File: docs/concepts_STEP.md [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+
+- **Interaction with multiple routing protocols:** Native support for OSPF and BGP protocols are available to interact with legacy routers, and provisioning is required to configure BGP over SVR between two 128T peers to exchange BGP routes. STEP provides an easy way to map the route information from these legacy routing protocols into the 128T world of services and service-routes, and provides an extensible way to interact with any future legacy protocols.  
+
+- **Multicast:** Receiver endpoints are advertised and the optimal replication point will be determined to send multicast packets to all receivers.
+
+- **Endpoint Availability:** Service routes can be configured for any service, but the availability of these service routes is not propagated across the network. STEP propagates the service availability information across the whole network so that each router can make intelligent decisions and minimize outages.
+
+- **Pinpoint Service Outages:** STEP stores the service and network information in the STEP Repository, providing a whole view of the network at any given moment. Service outages or heat-map views of the whole network can easily be generated from the repository.
```
Comment (MichaelBaj) [inline docs/concepts_STEP.md]: Will need to confirm with Markus and Abilash, but this capability will not be present in the first release of STEP.

**Pull request closed without merging**